src/api/layer.js

```diff
@@ -376,12 +376,21 @@ export default class Layer {
     }
 
     _addToMGLMap(map, beforeLayerID) {
-        if (map.isStyleLoaded()) {
+        this._isSourceLoaded = false;
+        
+        map.on('sourcedata', this._onMapSourcedata);
+        map.on('load', this._onMapLoad.bind(this, map, beforeLayerID));
```

Comment:
I think 

```js
map.on('load', this._onMapLoad.bind(this));
```
is more intuitive since we only bound the context but the parameters